service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/ServiceClient.java

```diff
@@ -76,9 +102,15 @@ protected ServiceClient(IotHubConnectionString iotHubConnectionString, IotHubSer
         this.userName = iotHubConnectionString.getUserString();
         this.sasToken = iotHubServiceSasToken.toString();
         this.iotHubServiceClientProtocol = iotHubServiceClientProtocol;
+        this.proxyOptions = proxyOptions;
+
+        if (this.proxyOptions != null && this.proxyOptions.getProxy() != null && this.iotHubServiceClientProtocol != IotHubServiceClientProtocol.AMQPS_WS)
```

Comment:
Is `this.proxyOptions.getProxy()` check required? There is only 1 public constructor for initializing the `ProxyOptions` class, and that takes in a `Proxy` object.

Reply:
Good point, just checking if proxyoptions is null should be sufficient here and everywhere else